src/Components/ChangePassword/ChangePassword.tsx

```diff
@@ -0,0 +1,145 @@
+import { fetchData } from "@/Redux/Features/Auth/ChangePasswordSlice";
+import { LockOpen } from "@mui/icons-material";
+import { LoadingButton } from "@mui/lab";
+import {
+  Button,
+  ListItemButton,
+  ListItemIcon,
+  ListItemText,
+  TextField,
+  Typography,
+} from "@mui/material";
+import Box from "@mui/material/Box";
+import Modal from "@mui/material/Modal";
+import { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { useDispatch, useSelector } from "react-redux";
+
+export default function ChangePassword() {
+  const dispatch = useDispatch();
+  const [open, setOpen] = useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+  const required = "This Field is required";
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    getValues,
+    formState: { errors },
+  } = useForm();
+  const onSubmit = (data: { email: string; password: string }) => {
+    dispatch(fetchData(data));
+  };
+  const { isConfirmPassword, loading } = useSelector(
+    (state) => state.changePassword
+  );
+  const [isPasswordChanged, setIsPasswordChanged] = useState(false);
+
+  useEffect(() => {
+    if (isConfirmPassword && !isPasswordChanged) {
```

Comment:
The logic here is incorrect, it will only work for the first time if I changed the password.
If you try to re-change your password, you'll notice that the effect will not run, because `isConfirmPassword` and `isPasswordChanged` will not change.
You need to make this logic dependent on the state of the request whether it was success or failure. If it was successs, then reset the form and close the modal. otherwise, we should show the user what is going wrong.